src/test/macromanager_test.cpp

```diff
@@ -0,0 +1,66 @@
+#include "recording/macromanager.h"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <QDebug>
+
+#include "mixxxtest.h"
+#include "signalpathtest.h"
+
+TEST(MacrosTest, CreateMacro) {
+    auto macro = new Macro();
+    ASSERT_EQ(macro->m_length, 0);
+}
+
+TEST(MacroManagerTest, ClaimRecording) {
+    auto mgr = MacroManager();
+    EXPECT_EQ(mgr.isRecordingActive(), false);
+    mgr.claimRecording();
+    EXPECT_EQ(mgr.isRecordingActive(), false);
+    mgr.m_macroRecordingState.store(MacroState::Armed);
+    mgr.claimRecording();
+    EXPECT_EQ(mgr.isRecordingActive(), true);
+}
+
+TEST(MacroManagerTest, RecordCueJump) {
+    auto mgr = MacroManager();
+    ChannelHandle handle = ChannelHandleFactory().getOrCreateHandle("");
+    EXPECT_EQ(mgr.m_macroRecordingState.load(), MacroState::Disabled);
+    mgr.notifyCueJump(handle, 0, 1);
+    EXPECT_EQ(mgr.m_activeChannel, nullptr);
+    mgr.m_macroRecordingState.store(MacroState::Armed);
+    mgr.notifyCueJump(handle, 0, 1);
+    EXPECT_EQ(mgr.m_activeChannel->handle(), handle.handle());
+    EXPECT_EQ(mgr.m_recordedMacro.actions[0].position, 0);
+    EXPECT_EQ(mgr.m_recordedMacro.actions[0].target, 1);
+    EXPECT_EQ(mgr.checkOrClaimRecording(handle), true);
+}
+
+TEST(MacroManagerTest, RecordingToggleControl) {
+    auto mgr = MacroManager();
+    ControlObject::set(ConfigKey(kMacroRecordingKey, "recording_toggle"), 1);
+    EXPECT_EQ(mgr.isRecordingActive(), true);
+    ControlObject::set(ConfigKey(kMacroRecordingKey, "recording_toggle"), 0);
+    EXPECT_EQ(mgr.isRecordingActive(), true);
+    ControlObject::set(ConfigKey(kMacroRecordingKey, "recording_toggle"), 1);
+    EXPECT_EQ(mgr.isRecordingActive(), false);
+}
+
+class MacroManagerE2ETest : public SignalPathTest {
+  public:
+    MacroManagerE2ETest()
+            : SignalPathTest(new MacroManager()) {
+    }
+};
+
+TEST_F(MacroManagerE2ETest, RecordSeek) {
+    auto mgr = m_pMacroManager;
```

Comment:
What are you trying to achieve here? Copy assignment? Creating and alias?

Reply:
an alias, to be consistent with the other tests. It's a pointer, so no copying anyways?

Reply:
I'd prefer to remove it and use `m_pMacroManager` directly. Otherwise I need to look up what `mgr` is and why it's aliased.